app/controllers/users_controller.rb

```diff
@@ -75,23 +74,71 @@ def photo
       type: @user.photo_content_type, disposition: :inline
   end
 
+  # GET users/1/change_email
+  def change_email
+  end
+
+  # GET users/1/change_password
+  def change_password
+  end
+
+  # PATCH users/1/update_email
+  def update_email
+    respond_to do |format|
+      if !@user.authenticate!(params[:user][:password], :password)
+        error_msg = t('.incorrect_password')
+        flash.now[:alert] = error_msg
+        format.html { render :change_email }
+        format.json { render json: {error: error_msg}, status: :unprocessable_entity }
+      elsif @user.update(email: params[:user][:email])
```

Comment:
Should we also permit this parameter for email update (as you've defined in L141 for password update)?

Reply:
Not in this case, as we are passing a specific value and not an unfiltered hash to update method.